Python/pystate.c

```diff
@@ -2009,42 +2016,57 @@ _Py_GetConfig(void)
 
 #define MINIMUM_OVERHEAD 1000
 
-PyObject **
-_PyThreadState_PushLocals(PyThreadState *tstate, int size)
+_Py_NO_INLINE PyObject **
+_PyThreadState_PushChunk(PyThreadState *tstate, int size)
+{
+    assert(tstate->datastack_top + size >= tstate->datastack_limit);
+
+    int allocate_size = DATA_STACK_CHUNK_SIZE;
+    while (allocate_size < (int)sizeof(PyObject*)*(size + MINIMUM_OVERHEAD)) {
+        allocate_size *= 2;
+    }
+    _PyStackChunk *new = allocate_chunk(allocate_size, tstate->datastack_chunk);
+    if (new == NULL) {
+        return NULL;
+    }
+    tstate->datastack_chunk->top = tstate->datastack_top - &tstate->datastack_chunk->data[0];
```

Comment:
```suggestion
    tstate->datastack_chunk->top = tstate->datastack_top - tstate->datastack_chunk->data;
```

Reply:
I am missing anything here?

Reply:
No. It's just a hint to the reader that `data` is an array within the chunk.
`chunk->data` looks like a pointer deference, but `&chunk->data[0]` hints that this is an address calculation.